shell/platform/darwin/macos/framework/Source/FlutterTextInputSemanticsObject.h

```diff
@@ -0,0 +1,89 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <Cocoa/Cocoa.h>
+
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterPlatformNodeDelegateMac.h"
+
+#include "flutter/third_party/accessibility/ax/platform/ax_platform_node_base.h"
+
+@class FlutterTextField;
+@class FlutterTextInputPlugin;
+
+namespace flutter {
+
+//------------------------------------------------------------------------------
+/// The ax platform node for a text field.
+class FlutterTextPlatformNode : public ui::AXPlatformNodeBase {
+ public:
+  //---------------------------------------------------------------------------
+  /// @brief      Creates a FlutterTextPlatformNode that uses a
+  ///             FlutterTextField as its NativeViewAccessible.
+  /// @param[in]  delegate          The delegate that provides accessibility
+  ///                               data.
+  /// @param[in]  view_controller   The view_controller that is used for querying
+  ///                               the information about FlutterView and
+  ///                               FlutterTextInputPlugin.
+  explicit FlutterTextPlatformNode(FlutterPlatformNodeDelegate* delegate,
+                                   __weak FlutterViewController* view_controller);
+  ~FlutterTextPlatformNode() override;
+
+  //------------------------------------------------------------------------------
+  /// @brief Gets the frame of this platform node relative to the view of
+  ///        FlutterViewController. This is used by the FlutterTextField to get its
+  ///        frame rect because the FlutterTextField is a subview of the
+  ///        FlutterViewController.view.
+  NSRect GetFrameRelativeToControllerView();
```

Comment:
I wonder if this could just be simplified to `GetFrame()`? In Apple terminology, the frame is in parent view co-ordinates whereas the bounds are in its own co-ordinate space.

@stuartmorgan is probably the right person to comment on that.